cfme/infrastructure/virtual_machines.py

```diff
@@ -960,22 +1066,21 @@ def get_number_of_vms(do_not_navigate=False):
 @navigator.register(Template, 'All')
 @navigator.register(Vm, 'All')
 class VmAllWithTemplates(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
-
-    def step(self, *args, **kwargs):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', '/vm_infra/explorer')
-        accordion.tree('VMs & Templates', 'All VMs & Templates')
+    VIEW = VmsTemplatesAllView
+    prerequisite = NavigateToObject(Server, 'LoggedIn')
 
-    def resetter(self, *args, **kwargs):
-        reset_page()
+    def step(self):
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Virtual Machines')
+        self.view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
 
-    def am_i_here(self, *args, **kwargs):
-        return match_page(summary='All VMs & Templates')
+    def am_i_here(self):
+        return self.view.is_displayed and self.view.sidebar.vmstemplates.is_opened
 
 
 @navigator.register(Template, 'AllForProvider')
 @navigator.register(Vm, 'AllForProvider')
 class VmAllWithTemplatesForProvider(CFMENavigateStep):
+    VIEW = VmsTemplatesAllView
```

Comment:
@mshriver but that leads to other independent views like AllArchived/AllOrphaned/AllForDatastore. maybe it would be better to add method inside VmsTemplatesAllView that would navigate thru accordion? this way we would get correct is_displayed from VmsTemplatesAllView. I tried to work with it creating am_i_here's in navigation steps - which were using is_displayed from VmsTemplatesAllView + additional checks as we were getting  "sub-views"